### PR TITLE
Enable warnings as errors

### PR DESCRIPTION
**Pull request by SSheldon**
Based off of #22, so that one should be merged first.

I noticed some more errors in the log output that we had missed. Treating warnings as errors is the best way to catch these :) Especially in Objective-C it makes sense to treat warnings as errors, because there are some very scary things that get reported only as a warning. We treat warnings as errors in the app.

I beefed up our warnings and fixed them, and now we can be sure more warnings won't be introduced.

### CODE REVIEW

File: Classes/Request/YLPGeoCoordinate.m [hunk 1]
```diff
@@ -31,15 +31,15 @@ - (instancetype)initWithLatitude:(double)latitude
...
     }
     
-    if (self.altitude) {
+    if (self.altitude != 0) {
         baseString = [self altitudeDescription:baseString];
     }
     
-    if (self.altitudeAccuracy) {
+    if (self.altitudeAccuracy != 0) {
         baseString = [self altitudeAccuracyDescription:baseString];
```
Comment (SSheldon) [inline Classes/Request/YLPGeoCoordinate.m]: @d9chen, are these `!= 0` checks correct? I don't totally understand why we would exclude these values if they were 0, but that's what the previous code is equivalent to.

Comment (d9chen) [inline Classes/Request/YLPGeoCoordinate.m]: @SSheldon I was actually under the assumption that the API returned error responses when 0 values were passed in so I was only adding them into the output string if they were provided. But I just did some searches with 0's for accuracy/altitude/altitudeAccuracy and they all worked fine, so let's just get rid of these checks, and put in the default 0 if they aren't provided.

**Comment by SSheldon** [comment]
Thanks for the guidance @d9chen, I removed those checks so that the accuracy & altitude are always included!

**Comment by d9chen** [comment]
:shipit: :)

**Pull request merged**